### PR TITLE
Support Rails 5.1

### PR DESCRIPTION
**Pull request by TylerRick**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,3 @@
-log
+/gemfiles/.bundle
+/log
+/vendor/bundle
```
Comment (grosser) [inline .gitignore]: if you add these to your global gitignore then you never need to add them to projects

File: .travis.yml [hunk 2]
```diff
@@ -1,9 +1,29 @@
...
- - gemfiles/rails4.gemfile
-script: bundle exec rake spec
+  - gemfiles/rails3.gemfile
+  - gemfiles/rails4.gemfile
+  - gemfiles/rails5.0.gemfile
+  - gemfiles/rails5.1.gemfile
+matrix:
+  exclude:
+    - gemfile: gemfiles/rails3.gemfile
+      rvm:
```
Comment (grosser) [inline .travis.yml]: FYI `rvm: [2.3, 2.4]`also works

**Comment by grosser** [comment]
ping me when it's green plz :)

File: .travis.yml (hunk 2 above)
Comment (TylerRick) [reply .travis.yml]: Much nicer, thanks!

File: .gitignore (hunk 1 above)
Comment (TylerRick) [reply .gitignore]: Good idea (done). Though we can't assume that anyone who checks out the project will have done so (and we don't want them `git add .`ing them by mistake).

**Comment by TylerRick** [comment]
@grosser Finally got all the jobs to pass.

Had to re-run it about 15 times for that to happen. Usually there were about 2 that failed with a network error like this:
```
fatal: unable to access 'https://github.com/grosser/ar_after_transaction.git/': Failed to connect to github.com port 443: Connection timed out
```
Travis CI seems unreliable.

**Comment by chamnap** [comment]
@grosser, please merge it. :)

**Pull request merged**